controllers/mustgather/template_test.go

```diff
@@ -16,29 +16,59 @@ import (
 )
 
 func Test_initializeJobTemplate(t *testing.T) {
-	testFailed := false
 	testName := "testName"
 	testNamespace := "testNamespace"
 	testServiceAccountRef := "testServiceAccountRef"
-	job := initializeJobTemplate(testName, testNamespace, testServiceAccountRef)
+	pvcClaimName := "test-pvc"
+	pvcSubPath := "test-path"
 
-	if got := job.Name; got != testName {
-		t.Logf("job name from initializeJobTemplate() was not correctly set. got %v, wanted %v", got, testName)
-		testFailed = true
+	tests := []struct {
+		name    string
+		storage *mustgatherv1alpha1.Storage
+	}{
+		{
+			name: "Without PVC",
+		},
+		{
+			name: "With PVC",
+			storage: &mustgatherv1alpha1.Storage{
+				Type: mustgatherv1alpha1.StorageTypePersistentVolume,
+				PersistentVolume: mustgatherv1alpha1.PersistentVolumeConfig{
+					Claim: mustgatherv1alpha1.PersistentVolumeClaimReference{
+						Name: pvcClaimName,
+					},
+					SubPath: pvcSubPath,
+				},
+			},
+		},
 	}
 
-	if got := job.Namespace; got != testNamespace {
-		t.Logf("job namespace from initializeJobTemplate() was not correctly set. got %v, wanted %v", got, testNamespace)
-		testFailed = true
-	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			job := initializeJobTemplate(testName, testNamespace, testServiceAccountRef, tt.storage)
 
-	if got := job.Spec.Template.Spec.ServiceAccountName; got != testServiceAccountRef {
-		t.Logf("job service account name from initializeJobTemplate() was not correctly set. got %v, wanted %v", got, testServiceAccountRef)
-		testFailed = true
-	}
+			if got := job.Name; got != testName {
+				t.Fatalf("job name from initializeJobTemplate() was not correctly set. got %v, wanted %v", got, testName)
+			}
+
+			if got := job.Namespace; got != testNamespace {
+				t.Fatalf("job namespace from initializeJobTemplate() was not correctly set. got %v, wanted %v", got, testNamespace)
+			}
 
-	if testFailed == true {
-		t.Error()
+			if got := job.Spec.Template.Spec.ServiceAccountName; got != testServiceAccountRef {
+				t.Fatalf("job service account name from initializeJobTemplate() was not correctly set. got %v, wanted %v", got, testServiceAccountRef)
+			}
+
+			if tt.storage != nil {
+				volume := job.Spec.Template.Spec.Volumes[0]
```

Comment:
```suggestion
                if len(job.Spec.Template.Spec.Volumes) == 0 {
 					t.Fatalf("expected at least one volume to be present")
				}
				volume := job.Spec.Template.Spec.Volumes[0]
```

similar to what we did for volumeMounts, to avoid panicking the test in case of a broken implementation.